src/components/MainNav/QueryStateUpdater.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { compose } from 'redux';
+import { injectIntl } from 'react-intl';
+import { withRouter } from 'react-router';
+import { isEqual, find } from 'lodash';
+
+import { withModules } from '../Modules';
+
+import {
+  updateQueryResource,
+  getLocationQuery,
+  updateLocation,
+  getCurrentModule,
+  isQueryResourceModule,
+  getQueryResourceState,
+} from '../../locationService';
+
+// onMount of stripes, sync the query state to the location.
+class QueryStateUpdater extends React.Component {
+  static propTypes = {
+    history: PropTypes.shape({
+      listen: PropTypes.func.isRequired,
+      replace: PropTypes.func.isRequired,
+      push: PropTypes.func.isRequired,
+    }).isRequired,
+    location: PropTypes.shape({
+      pathname: PropTypes.string,
+    }).isRequired,
+    modules: PropTypes.shape({
+      app: PropTypes.arrayOf(PropTypes.object),
+    }),
+    queryClient: PropTypes.shape({
+      removeQueries: PropTypes.func.isRequired,
+    }).isRequired,
+    stripes: PropTypes.shape({
+      store: PropTypes.shape({
+        subscribe: PropTypes.func.isRequired,
+      }),
+    }),
+  }
+
+  constructor(props) {
+    super(props);
+    this.store = props.stripes.store;
+  }
+
+  componentDidMount() {
+    let curQuery = getLocationQuery(this.props.location);
+    const prevQueryState = {};
+
+    this._unsubscribe = this.store.subscribe(() => {
+      const { history, location } = this.props;
+      const module = this.curModule;
+      const state = this.store.getState();
+
+      // If user has timed out, force them to log in again.
+      if (state?.okapi?.token && state.okapi.authFailure
+        && find(state.okapi.authFailure, { type: 'error', code: 'user.timeout' })) {
+        this.returnToLogin();
+      }
```

Comment:
This condition will never evaluate to true >= Sunflower because in RTR the token is only in the HTTPOnly cookie, so I guess it doesn't matter that `this.returnToLogin()` is undefined. IOW, this block looks like cruft to me and I think we should just remove it.

Reply:
Oh right! We removed this in original refactor... gonna do...